python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -1247,6 +1261,7 @@ def execute_asset_backfill_iteration_inner(
 
         updated_materialized_subset = AssetGraphSubset()
         failed_and_downstream_subset = AssetGraphSubset()
+        next_latest_storage_id = instance_queryer.instance.event_log_storage.get_maximum_record_id()
```

Comment:
It's ok for us to rely on the max record ID here because at this point the backfill hasn't launched any runs, so it's not possible to skip materialization events